huaweicloud/services/smn/data_source_huaweicloud_smn_topic.go

```
@@ -0,0 +1,160 @@
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"topic_urn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"display_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enterprise_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"push_policy": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceTopicRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.SmnV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SMN client: %s", err)
+	}
+
+	allTopics, err := topics.List(client).Extract()
+	if err != nil {
+		return diag.Errorf("unable to list topics: %s ", err)
+	}
+
+	log.Printf("[DEBUG] Retrieved SMN topics: %#v", allTopics)
+
+	filter := map[string]interface{}{
+		"TopicUrn":            d.Get("topic_urn"),
+		"Name":                d.Get("name"),
+		"EnterpriseProjectId": d.Get("enterprise_project_id"),
+		"DisplayName":         d.Get("display_name"),
+	}
+
+	filterTopics, err := utils.FilterSliceWithField(allTopics, filter)
+	if err != nil {
+		return diag.Errorf("filter topics failed: %s", err)
+	}
+
+	ids := make([]string, len(filterTopics))
+	stateTopics := make([]map[string]interface{}, len(filterTopics))
+	// create tagClient to fetch tags
+	tagClient, err := config.SmnV2TagClient(region)
+	if err != nil {
+		return diag.Errorf("error creating SMN tag client: %s", err)
```

should we consider to suppress this error as tags is an extra attribute of the topic?

---

getting tags error is suppressed, no need to suppress the error of creating client